File: InsuraTech.Domain/Events/PolicyExpiringSoonEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InsuraTech.Domain.Common;

namespace InsuraTech.Domain.Events
{
    public sealed class PolicyExpiringSoonEvent : IDomainEvent
    {


        public Guid EventId { get; } = Guid.NewGuid();
        public Guid Id => EventId;
        public DateTime OccurredOn { get; } = DateTime.UtcNow;
        public string EventType => nameof(PolicyExpiringSoonEvent);

        public Guid PolicyId { get; }
        public string PolicyNumber { get; }
        public string InsuredFullName { get; }
        public DateOnly ExpirationDate { get; }
        public int DaysRemaining { get; }

        public PolicyExpiringSoonEvent(Guid policyId, string policyNumber, string insuredFullName, DateOnly expirationDate, int daysRemaining)
        {
            PolicyId = policyId;
            PolicyNumber = policyNumber;
            InsuredFullName = insuredFullName;
            ExpirationDate = expirationDate;
            DaysRemaining = daysRemaining;
        }



    }
}

File: backend_v2/src/main/java/com/insuratech/domain/events/PolicyExpiringSoonEvent.java
package com.insuratech.domain.events;

import com.insuratech.domain.common.IDomainEvent;

import java.time.Instant;
import java.time.LocalDate;

public record PolicyExpiringSoonEvent(String policyId, String policyNumber, LocalDate expirationDate, Instant occurredOn) implements IDomainEvent {
    public PolicyExpiringSoonEvent(String policyId, String policyNumber, LocalDate expirationDate) {
        this(policyId, policyNumber, expirationDate, Instant.now());
    }
}
